pkg/oauthserver/oauthserver/oauth_apiserver.go

```diff
@@ -74,6 +75,12 @@ func internalOAuthConfigToExternal(oauthConfig configapi.OAuthConfig) (*osinv1.O
 func NewOAuthServerConfig(oauthConfig osinv1.OAuthConfig, userClientConfig *rest.Config) (*OAuthServerConfig, error) {
 	// TODO: there is probably some better way to do this
 	decoder := codecs.UniversalDecoder(osinv1.GroupVersion)
+	if len(oauthConfig.IdentityProviders) == 0 {
```

Comment:
Move this change right above the (that is the only point where you know the final length of the IDPs):

`ret := &OAuthServerConfig{`

The important part to understand is that there should be no encoding or decoding - you are not transferring data across components.  See how the `BootstrapIdentityProvider` is used below.

Also no pointless helper functions, inline the struct.

Reply:
got it, thanks